Latest dask (2.1.0) breaks _add_logging compute features

When starting up a dask job and wish to use `eliot.dask.compute_with_trace()` the job, the compute job initialization will fail to start.

```
# File eliot/dask.py

Line 130
>       assert result.keys() == dsk.keys()
E       AssertionError
```
In dask 1.2.1, the contents of the result and job keys were a list of tuples `tests/test_dask.py dict_keys([('from_sequence-ff4946f0379cbe77abc9f4e8013168b7', 0), ('lambda-e084aa13e4280bf986532e8292bfdff4', 0), ...` 

Whereas in dask 2.1.0, they're just a list of strings: `dict_keys(['combine_results-388d2de3-0346-4a09-a5b9-2e14d3a5f782', 'output_results-db913fd4-ce52-4f8b-8cbd-95cbde3915bb',...])`

 So the equality check no longer works.

Simple fix is to have the assert do a `set()` operation on both sides of the `==` sign, but it is curious why the keys themselves report themselves now as tuples instead of strings because it's not apparent where dask changed under its covers to cause the graph to change structure so much.

--- a/docs/source/news.rst
+++ b/docs/source/news.rst
@@ -4,9 +4,10 @@ What's New
 1.10.0
 ^^^^^^
 
-Bug fix:
+Bug fixes:
 
 * ``@eliot.testing.capture_logging`` now passes ``*args`` and ``**kwargs`` to the wrapped function, as one would expect. Fixes #420. Thanks to Jean-Paul Calderone for the bug report.
+* Eliot works with Dask 2.0. Thanks to Dan Myung for the bug report.
 
 1.9.0
 ^^^^^

--- a/eliot/dask.py
+++ b/eliot/dask.py
@@ -127,7 +127,7 @@ def _add_logging(dsk, ignore=None):
         )
         result[key] = (wrapped_func,) + tuple(args)
 
-    assert result.keys() == dsk.keys()
+    assert set(result.keys()) == set(dsk.keys())
     return result
 
 

--- a/tox.ini
+++ b/tox.ini
@@ -20,7 +20,7 @@ commands =
 [testenv:py35-numpy]
 basepython = python3.5
 deps = numpy
-       dask[bag]
+       dask[bag]==1.2.1
 
 [testenv:py36]
 basepython = python3.6
@@ -30,6 +30,7 @@ deps = cffi
 [testenv:py37]
 basepython = python3.7
 deps = cffi
+       dask[bag]
 
 [testenv:py35-twisted-latest]
 basepython = python3.5

